src/base/static/map-providers/abstract-map-factory.js

```diff
@@ -0,0 +1,213 @@
+const notImplemented = () => {
+  throw "Error: Method is not implemented for this mapping provider";
+};
+
+const abstractMethods = {
+  /**
+   * Create a new map instance.
+   * @interface
+   *
+   * @param {string} container - The id of the DOM element that will hold the map.
+   *
+   * @param {Object} [options] - Generic map options applicable to all providers.
+   * @param {number} [options.minZoom] - Minimum map zoom level allowed.
+   * @param {number} [options.maxZoom] - Maximum map zoom level allowed.
+   * @param {boolean} [options.attributionControl] - Whether to add an attribution to the map.
+   * @param {Object} [options.maxBounds] - Maximum bounds extent allowed.
+   * @param {number[]} [options.maxBounds.sw] - Southwest corner of allowed bounds, in [latitude, longitude] coordinates.
+   * @param {number[]} [options.maxBounds.ne] - Northeast corner of allowed bounds, in [latitude, longitude] coordiantes.
+   * @param {number} [options.zoom] - The initial zoom level of the map.
+   *
+   * @returns {Object} A reference to the map abstraction.
+   */
+  createMap: function(/*container, options, vendorOptions*/) {
+    notImplemented();
+  },
```

Comment:
Does this method of documentation and fallback to `notImplemented()` make sense?